domain/src/main/java/org/oppia/domain/question/QuestionTrainingController.kt

```diff
@@ -33,16 +34,45 @@ class QuestionTrainingController @Inject constructor(
    * @return a one-time [LiveData] to observe whether initiating the play request succeeded.
    * The training session may still fail to load, but this provides early-failure detection.
    */
-   fun startQuestionTrainingSession(skillIdsList: List<String>): LiveData<AsyncResult<Any?>> {
+  fun startQuestionTrainingSession(skillIdsList: List<String>): LiveData<AsyncResult<List<Question>>> {
     return try {
-        val questionsList = retrieveQuestionsForSkillIds(skillIdsList)
-        questionAssessmentProgressController.beginQuestionTrainingSession(questionsList)
-        MutableLiveData(AsyncResult.success<Any?>(null))
+      val retrieveQuestionsDataProvider = retrieveQuestionsForSkillIds(skillIdsList)
+      questionAssessmentProgressController.beginQuestionTrainingSession(
+        retrieveQuestionsDataProvider
+      )
+      dataProviders.convertToLiveData(retrieveQuestionsDataProvider)
     } catch (e: Exception) {
       MutableLiveData(AsyncResult.failed(e))
     }
   }
 
+  private fun retrieveQuestionsForSkillIds(skillIdsList: List<String>): DataProvider<List<Question>> {
+    val questionsDataProvider = topicController.retrieveQuestionsForSkillIds(skillIdsList)
+    return dataProviders.transform(TRAINING_QUESTIONS_PROVIDER, questionsDataProvider) {
+      getFilteredQuestionsForTraining(
+        skillIdsList, it.shuffled(Random(questionTrainingSeed)),
```

Comment:
Won't this cause the same order for subsequent assessments? I'm guessing we need to actually store the random object in a field to make sure it's shared across multiple shuffles.

Reply:
Yes that's right my mistake. I am storing the random object as a member variable and reusing the same object